### PR TITLE
Chore: Add appveyor CI

### LINKED ISSUE #123: CI: Add AppVeyor for Windows.
Our tests _should_ work, but untested so far.

Side note:

* `>>` redirection should work: https://www.microsoft.com/resources/documentation/windows/xp/all/proddocs/en-us/redirection.mspx?mfr=true

### PR DESCRIPTION
**Pull request by ryan-roemer**
* Fixes #123 
* Copies `process.env` and mutates a copy everywhere.
* Fixes previously undiscovered bugs with expand archetypes on Windows

/cc @exogen

### CODE REVIEW

File: stdout-%npm_package_config__proc_num%.log [hunk 1]
```diff
@@ -0,0 +1,2 @@
+string - from base
+string - from base
```
Comment (exogen) [inline stdout-%npm_package_config__proc_num%.log]: Hm, was adding this file intentional, and if so – what's up with it?

**Comment by exogen** [comment]
Amazing work so far, looks like it's almost there.

File: stdout-%npm_package_config__proc_num%.log (hunk 1 above)
Comment (ryan-roemer) [inline stdout-%npm_package_config__proc_num%.log]: haha, woops.

**Comment by exogen** [comment]
💯 LGTM

**Pull request merged**